internal/grpc/services/gateway/storageprovider.go

```diff
@@ -1028,6 +1030,12 @@ func (s *svc) handleCS3Ref(ctx context.Context, opaque string) (*provider.Resour
 	return res.Info, nil
 }
 
+func (s *svc) handleWebdavRef(ctx context.Context, ri *provider.ResourceInfo) (*provider.ResourceInfo, error) {
+	// A webdav ref has the following layout: <storage_id>/<opaque_id>@webdav_endpoint
+	// TODO: Once file transfer functionalities have been added.
```

Comment:
We still need to access the remote data by issues a GET/PROPFIND response to obtain the necessary information dynamically.  

We can merge this PR and add that later. @ishank011

Reply:
Yep, I added the TODO there. Once we have that functionality, I'll update this.